DecipherNow/gm-fabric-dashboard
# refactor: change state management to better render units dynamically

Resolves #1478 

## Things I may need to do
* I noticed that it seems like the lifecycle hooks are a bit messed up when I messed around in storybook. Something is probably off with how I'm storing local state and comparing things. 
* The Y axis also really should render the correct units
* I also need some realistic unit test around the new utility functions
  
## How to Test

  - Click around the app and look for regressions. Notice how the unit labels now render
  - Carefully review my crazy change
